src/components/badges/List.tsx

```diff
@@ -67,33 +57,16 @@ function BadgeListSuspended() {
       <ConfettiIfNeeded />
       <Scrollbar>
         <div className={space('space-y-2')}>
-          <BadgeSection
-            title="Mainnet NFT derivatives"
-            minted={ownedExternalERC721DerivativeContracts}
-            proofs={proofsAvailableToMint.filter(
-              (proof) =>
-                proof instanceof ERC721Proof &&
-                proof.network === Network.Mainnet
-            )}
-            onMinted={onMinted}
-          />
-          <BadgeSection
-            title="Goerli NFT derivatives"
-            minted={ownedERC721DerivativeContracts}
-            proofs={proofsAvailableToMint.filter(
-              (proof) =>
-                proof instanceof ERC721Proof && proof.network === Network.Goerli
-            )}
-            onMinted={onMinted}
-          />
-          <BadgeSection
-            title="Email derivatives"
-            minted={ownedEmailDerivativeContracts}
-            proofs={proofsAvailableToMint.filter(
-              (proof) => proof instanceof EmailProof
-            )}
-            onMinted={onMinted}
-          />
+          {Object.entries(ownedContractsByKey).map(([key, contracts]) => (
+            <>
+              <BadgeSection
+                title={data[key as keyof typeof data].title}
```

Comment:
You're doing `as keyof typeof data` four times in your PR, let's place this into `data` as type:
`DataContractNames`, for example, or `DataKeys`

Reply:
And let's use this type across the PR